bika/lims/browser/sample/view.py

```diff
@@ -380,32 +388,37 @@ def folderitem(self, obj, item, index):
         obj = item['obj']
 
         item['replace']['getSampleID'] = "<a href='%s'>%s</a>" % \
-            (item['url'], obj.getSampleID())
+            (item['url'], obj.getSampleID)
+        analysis_requests_brains = arc(portal_type='AnalysisRequest', UID=obj.getAnalysisRequestsUID)
+        # From Professional Plone 4 development:
+        # The getURL() method returns, as with the absolute_url() method, the current referenced
+        # object URL. This may be different from the server URL at the time that the object was indexed
         item['replace']['Requests'] = ",".join(
-            ["<a href='%s'>%s</a>" % (o.absolute_url(), o.Title())
-             for o in obj.getAnalysisRequests()])
-        item['Client'] = obj.aq_parent.Title()
+            ["<a href='%s'>%s</a>" % (o.getURL(), o.Title)
+             for o in analysis_requests_brains])
+        item['Client'] = obj.getClientTitle
         if hideclientlink == False:
+            head, tail = os.path.split(obj.getURL())
             item['replace']['Client'] = "<a href='%s'>%s</a>" % \
-                (obj.aq_parent.absolute_url(), obj.aq_parent.Title())
-        item['Creator'] = self.user_fullname(obj.Creator())
+                (head, obj.getClientTitle)
```

Comment:
Why don't you use `obj.getURL()` directly instead of `head` value?

afak, `obj.getURL()` will return the full URL for that object and it shouldn't be necessary to get rid of the last path component (`tail`). It would make sense if the url returned by `obj.getURL` was something like `http://example.com/lims/client-1/S-001/edit` (so you'd need to remove the `/edit` part), but I guess that this is not the case. In fact, stripping the url will redirect you to the URL that refers to the client instead.

Reply:
@xispa maybe I'm wrong but in this particular case (see snippet below) we are actually looking for the client url so that clicking on the client's name redirects the user to the client's url. This is why we get rid of the part of the url (```tail```) that refers to the sample. 

```python
if hideclientlink == False:
    head, tail = os.path.split(obj.getURL())
    item['replace']['Client'] = "<a href='%s'>%s</a>" % \
        (head, obj.getClientTitle)
```